central/image/datastore/store/common/v2/split_v2.go

```diff
@@ -124,3 +124,35 @@ func GenerateImageComponentV2(os string, image *storage.Image, from *storage.Emb
 
 	return ret, nil
 }
+func GenerateImageComponentV2FromImageV2(os string, image *storage.ImageV2, from *storage.EmbeddedImageScanComponent) (*storage.ImageComponentV2, error) {
```

Comment:
No need to add this, you can use `GenerateImageComponentV2` from here
https://github.com/stackrox/stackrox/blob/master/central/imagev2/datastore/store/common/split.go#L95